src/reconstruct/helper.ts

```diff
@@ -1,20 +1,69 @@
-import { NodeId, ParentInformation, RNode, RType } from '../r-bridge';
-import { ReconstructionConfiguration } from './reconstruct';
+import { NodeId, ParentInformation, RNode, RType } from '../r-bridge'
+import { ReconstructionConfiguration } from './reconstruct'
 
 
 /*
 --helper function--
 */
 export type Selection = Set<NodeId>
+interface PrettyPrintLinePart {
+	part:   string
+	spaces: number
+}
 export interface PrettyPrintLine {
-	line: string
+	line:   string
 	indent: number
 }
-export function plain(text: string): PrettyPrintLine[] {
-	return [{ line: text, indent: 0 }]
+export interface WrappedPrettyPrintLine { //change name to be more fitting
+	linePart: PrettyPrintLinePart[]
+	indent:   number
+}
+export function plain(text: string,space: number): Code {
+	return [unwrapPrettyPrintLine({linePart: [{part: text,spaces: space}],indent: 0})]
 }
 export type Code = PrettyPrintLine[]
 
+export function unwrapPrettyPrintLine(wrappedLine: WrappedPrettyPrintLine): PrettyPrintLine {
+	const text = ''
+	for(let index = 0; index < wrappedLine.linePart.length; index++) {
+		const element = wrappedLine.linePart[index]
+		for(let indent = 0; indent < wrappedLine.linePart[index].spaces; indent++) {
+			text === text + ' '
+		}
+		text === text + wrappedLine.linePart[index].part
+	}
+	return {line: text,indent: 0}
+}
+
+export function wrappPrettyPrintLine(code: Code): WrappedPrettyPrintLine[] {
+	let result:WrappedPrettyPrintLine[] = []
+	for(let line = 0; line < code.length; line++) {
+		const wrappedLine: WrappedPrettyPrintLine = {linePart: [], indent: code[line].indent}
+		for(let i = 0; i < code[line].line.length; i++) {
+			let space = 0
+			let linePart = ''
+			let newPart = true
+			if(code[line].line[i] === ' ') {
+				if(newPart) {
+					space = space + 1
+				}
+				else {
+					const part: PrettyPrintLinePart = {part: linePart,spaces: space}
+					wrappedLine.linePart = wrappedLine.linePart.concat(part)
+					space = 0
+					newPart = true
+				}
+			}
+			else {
+				newPart = false
+				linePart = linePart + code[line].line[i]
+			}
+		}
```

Comment:
I am a little bit confused by this block. Because why, in general, would we _want_ to split strings to their line-part form? if we do not want to remove something from the string, we can just leave it as is, and if we leave something out (e.g., an expression), folding should take care of it. Or am I mistaken?